### ISSUE TITLE
SelectItemIndexConverter getAsObject should test null on the submitted value

### ISSUE THREAD

**Issue by xqvier** [omnifaces/omnifaces#436]
All is in the title.

On WAS 8.5.5.11, the getAsObject method is called even when the submittedValue is null. 
This is the case when using a `<f:selectItem noSelectionOption="true" />`.
The implem used is myfaces 2.0.2

The method should test beforehand : 
``` 
if (pValue == null || pValue.isEmpty()) {
    return null;
}
```

**Comment by BalusC** [comment]
With SelectItemsIndexConverter, an empty submitted value in getAsObject() is totally unexpected as the getAsString() never returns an empty string. It always returns the index value, also for null item values.

Try a newer MyFaces version to exclude one and other.

**Comment by xqvier** [comment]
Actually, I'm forced to use this myfaces version as it is the one shipped with Websphere Application Server 8.5.5.11 ND (and still the one in the latest 8.5.5.13 but i have to be strictly compatible to the 8.5.5.11) and upgrading WAS provided library is for me a struggle (i'm still trying as there is a memory leak issue in that myfaces version). 

I know this is a myfaces 2.0.2 issues and that it seems resolved on later 2.0.x release (just guessing from the source code) but I opened this issue not because it's blocking us (I overrode it anyway) but just trying to make your library fill that compatibility matrix as it's really usefull and people could benefits from it. I've noticed in some part of your library that you fix some other library issue within your code without telling us to upgrade that said library :p 
If you're filling lazy I could also maybe do a MR but I think you'd lose more time telling me the comment isn't exactly as you like it than actually making the change. 
You can close it if you consider it out of scope. (no hard feelings :) 

As for the ` noSelectionOption`, I _think_ I'm using in its right context : using JSF2 and specifying an option that means no-option has been selected (which should result in a null in the backing bean)

**Comment by BalusC** [comment]
Added to 2.6.9-SNAPSHOT
